hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapFetchIndexOperation.java

```diff
@@ -130,20 +122,43 @@ protected void runInternal() {
         }
     }
 
+    @Nonnull
+    public static InternalIndex getInternalIndex(@Nonnull MapContainer mapContainer,
+                                                 @Nonnull String mapName,
+                                                 @Nonnull String indexName) {
+        Indexes indexes = mapContainer.getIndexes();
+        if (indexes == null) {
+            throw QueryException.error(SqlErrorCode.INDEX_INVALID, "Cannot use the index \"" + indexName
+                    + "\" of the IMap \"" + mapName + "\" because it is not global "
+                    + "(make sure the property \"" + ClusterProperty.GLOBAL_HD_INDEX_ENABLED
```

Comment:
Do we have plans to support local indices here?

Reply:
as part of https://hazelcast.atlassian.net/browse/HZ-2883 we probably will make this operation also working for local index (likely single partition at a time)